### DISCUSSION TITLE
June/July release

### DISCUSSION THREAD

**Discussion by pncnmnp** [pncnmnp/phoenix10.1#22] [Ideas]
As briefly mentioned in post #21, I believe that our next course of action should be to improve upon the existing state of this repository, before adding new features.

For instance, I would like to discuss the existing state of the "News" and "On This Day" actions.

# News

I absolutely despise using this feature! It lacks coherence and requires my full attention to comprehend the news - which defeats the purpose of broadcasting news.

In my experience, there is a significant difference between reading news and listening to it. When we listen to something, we need pauses, fillers, distinct and slow speech, and something to prevent our brains from zoning out.

My proposed solution is to do what Google Assistant does - when a user asks for news, download and integrate a daily news podcast into the feed. There are excellent daily news podcasts like PBS NewsHour, This Daily Podcast, and NBC Nightly News, with varying durations. For instance, the PBS one might be an hour long, and the NBC one might be 20 minutes. We can estimate the user's desired news duration and play the nearest news broadcast that meets their selection.

This may not be a perfect solution, as users may not receive news on niche topics like healthcare, but we can allow users to specify and modify categories to make it more customizable.

As I mentioned before, we do not want users to space out while listening to the news section. This will ensure that the news action remains engaging and interactive.

# On this day

This feature is implemented poorly, with the biggest issue being that Wikipedia's response is abrupt and better suited for reading than listening.

There are two options for moving forward: 
1. Axe the feature altogether, which is the simpler solution. 
2. Collect 365 fun facts from Wikipedia (one for each day), add them to ChatGPT-3.5-turbo, and ask for elaboration to make them more enjoyable to listen to.

Would love to know your thoughts on this!

**Comment by pncnmnp** [comment]
@ology Would love to know your thoughts on the above mentioned ideas.

**Comment by ology** [comment]
Re: News: I **definitely** like your idea for downloading an actual, daily news podcast for the `news` functionality. (And I need to play with the podcast functionality, as I have not yet, I confess.)

I do think that the RSS functionality is great, although limited. I'd rename it to just "rss" maybe? And then add a new schema category for the new "news" stuff?

Sorry if I'm missing something! It's the weekend and I can finally focus on the important things! Haha :D

**Reply by pncnmnp** [reply]
No worries! :) 

Definitely! We can keep the RSS functionality as it is and assign the `news` action to our daily news podcast idea. Let me write some code and see how it goes.

**Comment by ology** [comment]
Re: On this day - Yes. I'd (tediously) collect GPT responses for each day. Other literal text is used, so why not this too. As with the news feature, I'd maybe keep the existing functionality and rename it so it's distinguished from the smoother method.

**Reply by pncnmnp** [reply]
Yup! I agree. I'll work on the "On This Day" feature too.

**Comment by ology** [comment]
So I wish I were a more competent python programmer. 'Cause I'd submit technical patches too. As it is, you obviously know up from down! :)

**Reply by pncnmnp** [reply]
No worries! 😄 Your feedback is super valuable, and it does help that I know the codebase quite well!

**Comment by ology** [comment]
Is the RSS feature really distinct from the podcast (or current news) feature? 🤔

**Reply by ology** [reply]
I will study...

**Reply by pncnmnp** [reply]
I understand that if we decide to keep both the RSS and current news features, they might be somewhat distinct.

Currently, we fetch RSS news from the New York Times. The New York Times has a daily 20-minute podcast called "The Daily". However, this podcast only focuses on a single news topic. Therefore, we might have to fetch news from something like "NBC Nightly News with Lester Holt" since it offers a much more holistic view. I'm sure that if we compare the top news from the RSS feeds to the news presented in the aforementioned podcast, there will be a lot of similarity.

RSS news is particularly useful when fetching news in topics like healthcare or sports. We do have "The Sports Daily with Reality Steve" and "Gist Healthcare Daily" podcasts, but I'm not entirely certain about the quality of the news they provide.

However, the biggest reason why I believe news podcasts might outshine RSS feeds-based content is that they are pleasant to listen to. The voice is not monotonous, and they bring in multiple guests at times, which prevents users from zoning out. I have been experiencing this problem a lot in the last six months.

**Reply by ology** [reply]
Ok understood. Thank you for the explanation.

Another thing about news podcasts vs rss is that traditionally they have been done by a news announcer and not the DJ. Speaking of which - If the rss news could be read in the AI voice of https://en.wikipedia.org/wiki/Les_Nessman that would be totally hilarious. But I digress...

**Reply by pncnmnp** [reply]
Hahaha 😄! I've been searching for a good open-source alternative to Coqui-ai's TTS for some time now. Currently, this is the best option available. Tortoise-tts is extremely slow - it may take a few minutes to create a single sentence and requires a good GPU, something I don't have. Mycroft-ai's TTS is another option, but it doesn't match the quality of Coqui-ai's offering.

The only remaining option is from ElevenLabs. However, as can be seen in my comment here https://github.com/pncnmnp/phoenix10.1/issues/8:
> They are saying that 10000 characters should give us about 10 minutes of voice over.

> My back-of-the-envelope calculation tells me that 10,000 characters should give us about 1500 words or 3-4 pages. One normal sized schema should take around 3/4th page (assuming that we synthesize news as well). So, we might be able to get 4-5 highly quality radio broadcasts out of this. Maybe we can then default to Coqui-ai.



Unfortunately, this is not sufficient to create daily podcasts. I am optimistic that by the end of this year or next year, we will have a reliable open-source TTS library that works well on CPUs. Once we have that, we can enter the Joe Rogan and Les Nessman territory.

**Reply by pncnmnp** [reply]
Looks like FB just open-sourced a TTS model - https://github.com/facebookresearch/fairseq/tree/main/examples/mms#tts

Not sure how good it is, but we should try and experiment with it.

**Comment by ology** [comment]
And what is "itunes_author"? I am trying out the podcast functionality and get:

```
Traceback (most recent call last):
  File "radio.py", line 967, in <module>
    dialogue.flow()
  File "radio.py", line 739, in flow
    speech = self.podcast_dialogue(rss_feed)
  File "radio.py", line 489, in podcast_dialogue
    "You know, I love listening to podcasts. "
KeyError: 'itunes_author'
```

Why do podcasts need to be indexed by itunes?

**Reply by pncnmnp** [reply]
> Why do podcasts need to be indexed by itunes?

They need to be indexed to obtain the metadata, similar to how we index songs. 

Could you please share the RSS feed of your podcast that you used in your schema? I have never encountered this issue before. We should implement a fallback condition.

**Reply by ology** [reply]
https://www.wired.com/feed/tag/ai/latest/rss

**Reply by pncnmnp** [reply]
Ah! They only have info for:
> 'title', 'episodes', 'description', 'link', 'language'

in their XML file. We should have a fallback condition! I'll fix this.

**Reply by ology** [reply]
Ha! You are The Man!

**Reply by pncnmnp** [reply]
https://github.com/pncnmnp/phoenix10.1/issues/23
